lib/miq_pglogical.rb

```diff
@@ -7,22 +7,41 @@ class MiqPglogical
   PUBLICATION_NAME = 'miq'.freeze
   ALWAYS_EXCLUDED_TABLES = %w(ar_internal_metadata schema_migrations repl_events repl_monitor repl_nodes).freeze
 
-  def self.with_connection_error_handling
-    retry_attempted ||= false
-    yield
-  rescue PG::ConnectionBad
-    raise if retry_attempted
-
-    pglogical(true)
-    retry_attempted = true
-    retry
-  end
+  module ConnectionHandling
```

Comment:
minor but we should move this into a `lib/miq_pglogical/connection_handling.rb` file (particularly for when we get to Zeitwerk)

Reply:
I changed it just to avoid any back and forth since I am gone tomorrow, but provided a rational below as to why I didn't do this originally ( since I am out tomorrow).

* * *

For the record:  The reason I don't like doing this is because I don't like modules that are nested under classes to be part of different files, since it means you can technically F-up autoloading by requiring the nested module first via a `require`, `MiqPglogical::ConnectionHandling`, and then when trying to call something on `MiqPglogical` itself, it is just an empty class.

Where this was just a one off module, I figured it was fine as a nested one, but maybe it should just be something like `module PgConnectionHandling`.  Otherwise, if I was doing this from scratch, I would say the `MiqPglogical` should be the "namespace 'module`", and the existing class should be something like `Base` or whatnot, and then `ConnectionHandling` can live as a nested module under that.